extensions/amp-story/1.0/test/test-amp-story-tooltip.js

```diff
@@ -0,0 +1,189 @@
+  });
+
+  it('should show the tooltip on store property update', () => {
+    fakePage.appendChild(clickableEl);
+
+    storeService.dispatch(Action.TOGGLE_TOOLTIP, clickableEl);
+
+    expect(tooltip.tooltipOverlayEl_).to.not.have.class('i-amphtml-hidden');
+  });
+
+  it('should hide the tooltip on store property update', () => {
+    fakePage.appendChild(clickableEl);
+
+    storeService.dispatch(Action.TOGGLE_TOOLTIP, clickableEl);
+    storeService.dispatch(Action.TOGGLE_TOOLTIP, null);
+
+    expect(tooltip.tooltipOverlayEl_).to.have.class('i-amphtml-hidden');
+  });
+
+  it('should hide the tooltip when switching page', () => {
+    fakePage.appendChild(clickableEl);
+    storeService.dispatch(Action.TOGGLE_TOOLTIP, clickableEl);
+
+    storeService.dispatch(Action.CHANGE_PAGE, {id: 'newPageId'});
+
+    expect(tooltip.tooltipOverlayEl_).to.have.class('i-amphtml-hidden');
+  });
+
+  it('should hide the tooltip when clicking outside of it', () => {
+    fakePage.appendChild(clickableEl);
+    storeService.dispatch(Action.TOGGLE_TOOLTIP, clickableEl);
+
+    tooltip.tooltipOverlayEl_.dispatchEvent(new Event('click'));
+
+    expect(tooltip.tooltipOverlayEl_).to.have.class('i-amphtml-hidden');
+  });
+
+  it('should navigate when tooltip is open and user clicks on arrow', () => {
+    fakePage.appendChild(clickableEl);
+    storeService.dispatch(Action.TOGGLE_TOOLTIP, clickableEl);
+    const fakeNavCall = sandbox.stub(tooltip, 'onNavigationalClick_');
+
+    const rightButton = tooltip.tooltipOverlayEl_
+        .querySelector('.i-amphtml-story-tooltip-layer-nav-button' +
+          '.i-amphtml-story-tooltip-nav-button-right');
+
+    rightButton.dispatchEvent(new Event('click'));
+
+    expect(fakeNavCall).to.have.been.calledOnce;
+  });
+
+  it('should navigate to previous page when clicked on right arrow and ' +
+    'story is RTL', () => {
+    fakePage.appendChild(clickableEl);
+    storeService.dispatch(Action.TOGGLE_RTL, true);
+    storeService.dispatch(Action.TOGGLE_TOOLTIP, clickableEl);
+
+    const fakeNavCall = sandbox.stub(tooltip, 'onNavigationalClick_');
+    const rightButton = tooltip.tooltipOverlayEl_
+        .querySelector('.i-amphtml-story-tooltip-layer-nav-button' +
+          '.i-amphtml-story-tooltip-nav-button-right');
+
+    rightButton.dispatchEvent(new Event('click'));
+
+    expect(fakeNavCall).to.be.calledWith(
+        sinon.match(event => event.type == 'click'),
+        EventType.PREVIOUS_PAGE);
+  });
+
+  it('should append icon when icon attribute is present', () => {
+    addAttributesToElement(clickableEl, {'data-tooltip-icon': '/my-icon'});
+    fakePage.appendChild(clickableEl);
+    storeService.dispatch(Action.TOGGLE_TOOLTIP, clickableEl);
+
+    const tooltipIconEl = tooltip.tooltipOverlayEl_
+        .querySelector('.i-amphtml-story-tooltip-icon').firstElementChild;
+
+    expect(tooltipIconEl).to.have.attribute('src');
+    expect(tooltipIconEl['src']).to.equal('http://localhost:9876/my-icon');
```

Comment:
Nit: `.getAttribute('src')`

Also, I believe the `9876` port can be changed (not sure?). 

`expect(tooltipIconEl.getAttribute('src')).to.equal(win.location.origin + '/my-icon');`

Reply:
`win.location.origin` doesn't work :/

But I've seen `localhost:9876` used throughout the codebase, so we should be fine:

https://github.com/ampproject/amphtml/search?q=http%3A%2F%2Flocalhost%3A9876&unscoped_q=http%3A%2F%2Flocalhost%3A9876